src/ObjCRuntime/Runtime.cs

```diff
@@ -308,6 +308,34 @@ static IntPtr GetFunctionPointer (Delegate d)
 			return Marshal.GetFunctionPointerForDelegate (d);
 		}
 
+		// value_handle: GCHandle to a (smart) enum value
+		// returns: a handle to a native NSString *
+		static IntPtr ConvertSmartEnumToNSString (IntPtr value_handle)
+		{
+			var value = GCHandle.FromIntPtr (value_handle).Target;
+			var smart_type = value.GetType ();
+			MethodBase getConstantMethod, getValueMethod;
+			if (!Registrar.IsSmartEnum (smart_type, out getConstantMethod, out getValueMethod))
+				throw ErrorHelper.CreateError (8024, $"Could not find a valid extension type for the smart enum '{smart_type.FullName}'. Please file a bug at https://bugzilla.xamarin.com.");
+			var rv = (NSString) ((MethodInfo) getConstantMethod).Invoke (null, new object [] { value });
```

Comment:
this can return `null`
in particular in cases where the field is not available on the OS version being executed